File: code/a3262.pas
var a : array[1..10000] of integer;
    n, i, v, d, dmax : integer;
BEGIN
    readln(n);
    for i:= 1 to n do read(a[i]);
    for i:= 1 to n do begin
        if a[i] mod 2 = 0 then begin
            inc(d);
            if d>dmax then begin
                dmax:=d;
                v:=i-d+1;
            end;
        end
        else
            d:=0;
    end;
    for i:= 0 to dmax-1 do
        write(a[i+v],' ');
END.

File: code/a3262.cpp
#include <iostream>
using namespace std;
int n, a[10000];
int main()
{
    cin>>n;
    int i, v, d=0, dmax=0;
    for (i=1; i<=n; i++) cin>>a[i];
    for (i=1; i<=n; i++) {
        if (a[i]%2==0) {
            d++;
            if (d>dmax) {
                dmax=d;
                v=i-d+1;
            }
        }
        else {
            d=0;
        }
    }
    for (i=0; i<dmax; i++)
        cout<<a[v+i]<<" ";
    return 0;
}